Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts

```diff
@@ -19,4 +36,108 @@ export class CalendarComponent {
       right: 'dayGridMonth,timeGridWeek,timeGridDay'
     }
   };
+
+
+  constructor(public dialog: MatDialog,
+              private toastr: ToastrService,
+              private calendarService: CalendarService) {
+  }
+
+  ngOnInit() {
+    this.calendarService.getEvents()
+      .subscribe(events  => {
+        this.events = events;
+        this.calendarOptions.events = this.events;
+      });
+  }
+
+  ngAfterViewInit() {
+    this.calendarComponentApi = this.calendarComponent.getApi();
+  }
+
+  onOpenDialog(): void {
+    const dialogRef = this.dialog.open(EventModalComponent, {
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
```

Comment:
Use `OnDestroy` event to unsubscribe from Observable

Reply:
fixed